index.html

```diff
@@ -2945,6 +3042,168 @@ <h3>
+        </h3>
+        <p>
+          The <dfn>description</dfn> member of a <a>ShortcutItem</a> is a
+          <a>string</a> that allows the developer to describe the purpose of
+          the shortcut. User agents may choose to expose this information to
+          assistive technology.
+        </p>
+      </section>
+      <section data-dfn-for="ShortcutItem" data-link-for="ShortcutItem">
+        <h3>
+          <code>url</code> member
+        </h3>
+        <p>
+          The <dfn>url</dfn> member of a <a>ShortcutItem</a> is a <a>URL</a> to
+          which a user agent should navigate when the associated shortcut is
+          activated.
+        </p>
+        <p>
+          The steps for <dfn>processing the <code>url</code> member of a
+          shortcut</dfn> are given by the following algorithm. The algorithm
+          takes a <a>ShortcutItem</a> <var>url</var>, and a <var>manifest
+          URL</var>, which is the <a>URL</a> from which the <var>manifest</var>
+          was fetched. This algorithm will return a <a>URL</a> or
+          <code>undefined</code>.
+        </p>
+        <ol>
+          <li>Let <var>value</var> be <var>url</var>
+          </li>
+          <li>If <a>Type</a>(<var>value</var>) it not <a>String</a>, return
+          <code>undefined</code>.
+          </li>
+          <li>Otherwise, <a>parse</a> <var>value</var> using <var>manifest
+          URL</var> as the base URL and return the result.
+          </li>
+        </ol>
+      </section>
+      <section data-dfn-for="ShortcutItem" data-link-for="ShortcutItem">
+        <h3>
+          <dfn>params</dfn> member
+        </h3>
+        <p>
+          The <a>params</a> member of an <a>ShortcutItem</a> is a
+          <a>RequestParams</a> object defining the keys and values that should
+          be added to the <var>url</var> of a shortcut before navigation
+          occurs. See also: <a>launching a shortcut</a>.
+        </p>
+      </section>
+      <section data-dfn-for="ShortcutItem" data-link-for="ShortcutItem">
+        <h3>
+          <dfn>icons</dfn> member
+        </h3>
+        <p>
+          The <a>icons</a> member of an <a>ShortcutItem</a> member is an
+          <a>array</a> of <a>ImageResource</a>s that can serve as iconic
+          representations of the shortcut in various contexts.
+        </p>
+      </section>
+      <section>
+        <h3>
+          <dfn>Launching a shortcut</dfn>
+        </h3>
+        <p>
+          When <a>ShortcutItem</a> <var>shortcut</var> having
+          <a>WebAppManifest</a> <var>manifest</var> is invoked, run the
+          following steps:
+        </p>
+        <ol>
+          <li>Let <var>url</var> be a copy of <var>shortcut.url</var>.
+          </li>
+          <li>Let <var>header list</var> be a new empty <a data-xref-for=
+          "request">header list</a>.
+          </li>
+          <li>If <var>shortcut.params</var> <a data-cite="infra/#list-contain">
```

Comment:
Why do we need `shortcut.params`? This doesn't let the user, user agent, or site dynamically inject parameters into the URL, does it? Like, it seems that the params are a static set of key/value pairs defined in the manifest alongside `url`. Thus, why do we need to specify this as a separate dictionary that gets encoded into a query string? Why not just let the manifest specify query parameters in the `url` directly?

As an example, say you had this:

```json
"shortcuts": [
  {
      "name": "Subscriptions",
      "url": "/subscriptions",
      "params": {
        "sortby": "ascending"
      }
  }
]
```

why not just express it as this:

```json
"shortcuts": [
  {
      "name": "Subscriptions",
      "url": "/subscriptions?sortby=ascending"
  }
]
```

Reply:
Fair point. I think this was a vestige of when we had considered supporting POST, etc. and could likely be done away with assuming everyone agrees that shortcuts should only support GET requests.

Reply:
Oh OK, I didn't know there was consideration to support POST. Now that you mention it, why was it decided not to? Is it "maybe in the future" or we have an active reason not to?

I agree it would be a little weird to click a shortcut and have it perform some side effect, and it's probably a better user experience to take you to a GET page that asks if you want to perform the side effect first. So I'm cool with disallowing POST but just wanted to get the rationale.